be/btree.c

```diff
@@ -816,18 +817,17 @@ static void be_btree_move_parent_key_to_right_child(struct m0_be_bnode *parent,
 
 	while (i > 0) {
 		rch->bt_kv_arr[i] = rch->bt_kv_arr[i - 1];
-		rch->bt_ik[i] = rch->bt_ik[i - 1];
-		if(ktype != M0_BBT_CAS_CTG)
-		{
-			rch->bt_kv_arr[i].btree_key = &rch->bt_ik[i].inlkey;
-		}
+		// if(ktype != M0_BBT_CAS_CTG)
+		// {
+		// 	rch->bt_kv_arr[i].btree_key = &rch->bt_ik[i].inlkey;
+		// }
 		rch->bt_child_arr[i + 1] = rch->bt_child_arr[i];
 		--i;
 	}
 	rch->bt_child_arr[1] = rch->bt_child_arr[0];
 	rch->bt_kv_arr[0] = parent->bt_kv_arr[idx];
-	rch->bt_ik[0] = parent->bt_ik[idx];
-	rch->allocated[0] = parent->allocated[idx];
+	*rch->bt_ik[0].inlkey = *((char *)parent->bt_kv_arr[idx].btree_key);
```

Comment:
You cannot use bt_ik[0] since it might already be allocated. You will need to get the free entry using the find routine.